docs/overview.md

```diff
@@ -0,0 +1,797 @@
+
+```
+d(((A,B),C),D) = MAX(d((A,B),D), d(C,D))
+d(((A,B),C),D) = MAX(8, 5)
+d(((A,B),C),D) = 8
+```
+
+```
+dists        ((A,B),C)    D
+((A,B),C)    0            8
+D            8            0
+```
+
+![](images/complete-linkage-A-B-C.png)
+
+Lastly, the next (and only remaining) smallest distance is `d(((A,B),C),D) = 8`, so we cluster those and then we've completed a hierarchical clustering using a complete-linkage clustering method:
+
+![](images/complete-linkage-full.png)
+
+The height of the above dendrogram (the y-axis labelled "Distance") tells us the greatest distance between samples within a clustering. For example, the node of `(C,(A,B))` has a height of `4`, which tells us that the `(C,(A,B))` cluster has a distance of `4`. This can have different meanings depending on the linkage method used to generate the linkage. Since we used a complete-linkage method to generate the linkage for this dendrogram, this tells us that the maximum distance between any two elements of the cluster is `4`. Further, since our distances are derived from the Hamming distance between categorical vectors of gene variants, this specifically tells us that within the `(C,(A,B))` cluster, no two samples have more than `4` gene variant (i.e. allele) differences between them.
+
+### Complete-Linkage Data Object
+
+GAS's mcluster uses SciPy's `scipy.cluster.hierarchy.linkage` function to generate a linkage object, which represents the hierarchical clustering performed in the previous step.
+
+```
+[[0. 1. 1. 2.]
+ [2. 4. 4. 3.]
+ [3. 5. 8. 4.]]
+```
+
+The above linkage has the following structure:
+
+|           | Cluster Index 1 | Cluster Index 2 | Distance | Number of Elements |
+| --------- | --------------- | --------------- | -------- | ------------------ |
+| Cluster 1 | 0               | 1               | 1        | 2                  |
+| Cluster 2 | 2               | 4               | 4        | 3                  |
+| Cluster 3 | 3               | 5               | 8        | 4                  |
+
+Each row represents a cluster in the hierarchical cluster. *Cluster Index 1* and *Cluster Index 2* refer to the index of the clusters that make up the cluster. For example, for Cluster 1, index `0` refers to cluster `A` and index `1` refers to cluster `B`, meaning the row represents the `(A,B)` cluster. This is explained in detail more below. *Distance* represents the distance of elements within clusters, which can have different meanings depending on the linkage method used. *Number of Elements* represents the number of elements (input samples) within the cluster.
+
+There are a number of implied clusters that are not shown in the linkage, but are implicitly referenced by the linkage. These are clusters with only one element: the original inputs. The linkage can be expanded to improve clarity:
+
+```
+[[0. 0. 0. 1.]
+ [1. 1. 0. 1.]
+ [2. 2. 0. 1.]
+ [3. 3. 0. 1.]
+ [0. 1. 1. 2.]
```

Comment:
Maybe add a space here to separate the first 4 rows from everything else

Reply:
307e48c